packages/backend/src/open_payments/payment/incoming/model.ts

```diff
@@ -239,39 +242,60 @@ export class IncomingPayment
     return payment
   }
 
-  public toOpenPaymentsType({
-    ilpStreamConnection
-  }: {
+  public toOpenPaymentsType(
+    paymentPointer: PaymentPointer
+  ): OpenPaymentsIncomingPayment
+  public toOpenPaymentsType(
+    paymentPointer: PaymentPointer,
     ilpStreamConnection: Connection
-  }): OpenPaymentsIncomingPayment {
-    return {
-      id: this.url,
-      paymentPointer: this.paymentPointer.url,
+  ): OpenPaymentsIncomingPaymentWithConnection
+  public toOpenPaymentsType(
+    paymentPointer: PaymentPointer,
+    ilpStreamConnection: string
+  ): OpenPaymentsIncomingPaymentWithConnectionUrl
+  public toOpenPaymentsType(
+    paymentPointer: PaymentPointer,
+    ilpStreamConnection?: Connection | string
+  ):
+    | OpenPaymentsIncomingPaymentWithConnection
+    | OpenPaymentsIncomingPaymentWithConnectionUrl
+
+  public toOpenPaymentsType(
```

Comment:
What if we added some `model.test.ts`s at least to test `toOpenPaymentsType`?
Or are existing response body checks in routes tests good enough?